### ISSUE TITLE
Revise macOS platform support after GitHub runner switched to Arm64

### ISSUE THREAD

**Issue by koyeung** [tikv/jemallocator#85]
Any idea to revise the platform support after both GitHub runner `macos-14` and `macos-latest` are now "macOS 14 Arm64"?

ref: https://github.com/actions/runner-images/blob/main/README.md

Currently in test workflow: https://github.com/tikv/jemallocator/blob/main/.github/workflows/main.yml
```yaml
- name: aarch64-apple-darwin
  target: aarch64-apple-darwin
  nobgt: 0
  no_tests: 1
  tag: macos-14
- name: x86_64-apple-darwin (stable)
  target: x86_64-apple-darwin
  nobgt: 0
  no_tests: 1
  tag: macos-latest
```

**Comment by BusyJay** [comment]
We may change macos-14 to macos-latest and change macos-latest to macos-13

**Comment by BusyJay** [comment]
Fixed by #86.
